third_party/mbedtls/CMakeLists.txt

```diff
@@ -39,7 +39,7 @@ target_compile_definitions(mbedtls
     PUBLIC
         "MBEDTLS_CONFIG_FILE=$<IF:$<BOOL:${OT_MBEDTLS_CONFIG_FILE}>,${OT_MBEDTLS_CONFIG_FILE},${OT_MBEDTLS_DEFAULT_CONFIG_FILE}>"
     PRIVATE
-        ${OT_PRIVATE_DEFINES}
+        $<TARGET_PROPERTY:ot-config,INTERFACE_COMPILE_DEFINITIONS>
```

Comment:
Why don't we use `target_link_libraries`?

Reply:
I forgot to mention, the mbedtls is using an old cmake version, which doesn't support this yet.